spec/services/task_service/gpt_generate_tests_spec.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe TaskService::GptGenerateTests do
+  describe '.new' do
+    subject(:gpt_generate_tests_service) { described_class.new(task:) }
+
+    let(:programming_language) { build(:programming_language, language: 'Ruby') }
+    let(:task) { build(:task, description: 'Sample Task', programming_language:) }
+
+    it 'assigns task' do
+      expect(gpt_generate_tests_service.instance_variable_get(:@task)).to be task
+    end
+
+    it 'assigns message from task description' do
+      expect(gpt_generate_tests_service.instance_variable_get(:@message)).to eq('Sample Task')
+    end
+
+    it 'assigns language from programming language' do
+      expect(gpt_generate_tests_service.instance_variable_get(:@language)).to eq('Ruby')
+    end
+  end
+
+  describe '#execute' do
+    let(:programming_language) { create(:programming_language, language: 'Python') }
+    let(:task) { create(:task, description: 'Create a Python script.', programming_language:) }
+    let(:gpt_generate_tests_service) { described_class.new(task:) }
```

Comment:
Also here I am not sure what I need to do 😬
should I add a similar line to this `subject(:convert_zip_to_tasks) { described_class.call(zip_file:) }` here?

Reply:
yes exactly. The method you want to test should be the `subject`. In this case it could look like this:

```ruby
describe '.call' do
    subject(:gpt_generate_tests) { described_class.call(task:) }
    let(:programming_language) { create(:programming_language, language: 'Python') }
    let(:task) { create(:task, description: 'Create a Python script.', programming_language:) }
```
an expectation could look like this
```ruby
expect { gpt_generate_tests }.to ....
```

Reply:
I am not sure if this is what you meant.

I have added this function as public:

```
    def process_gpt_response
      gpt_response
    end
```

and adjusted the tests as follows as you suggested:


```
  describe '#gpt_response' do
    subject(:gpt_generate_tests_service) { described_class.new(task:) }

    let(:programming_language) { build(:programming_language, language: 'Ruby') }
    let(:task) { build(:task, description: 'Sample Task', programming_language:) }
    let(:mock_client) { instance_double(OpenAI::Client) }

    before do
      allow(OpenAI::Client).to receive(:new).and_return(mock_client)
    end

    it 'raises InvalidTaskDescription when response does not contain backticks' do
      allow(mock_client).to receive(:chat).and_return({'choices' => [{'message' => {'content' => 'Sample response'}}]})
      expect { gpt_generate_tests_service.process_gpt_response }.to raise_error(InvalidTaskDescription)
    end

    it 'extracts code from response when enclosed in backticks' do
      allow(mock_client).to receive(:chat).and_return({'choices' => [{'message' => {'content' => "```ruby\ndef test_sample\n  assert true\nend\n```"}}]})
      expect(gpt_generate_tests_service.process_gpt_response).to eq("def test_sample\n  assert true\nend")
    end
  end
```

Is this what you were referring to? :)

Reply:
We discussed that in person today.

Reply:
I had a meeting with Sebastian and this should be fixed now. please take a look whenever possible to resolve the comment. thanks a lot for your feedback and help 😊

Reply:
A lot of steps in the right direction, I added a new suggestions for the last steps I had in mind.